drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/strategies/AbstractDataManagementStrategy.java

```diff
@@ -64,9 +64,17 @@ public static FactModelTree getSimpleClassFactModelTree(String simpleClass, Stri
         return toReturn;
     }
 
+    /**
+     * It adds an "expression" simple property to the given <code></code>
+     * @param factModelTree
+     */
     public void populateExpressionProperty(FactModelTree factModelTree) {
-        factModelTree.addSimpleProperty(factModelTree.getFactName() + ConstantsHolder.EXPRESSION_SUFFIX,
-                                        ConstantsHolder.EXPRESSION_CLASSNAME);
+        if (factModelTree.getSimpleProperties().containsValue(ConstantsHolder.EXPRESSION_CLASSNAME)) {
+            return;
+        }
+        final String expressionName = factModelTree.getFactName() + ConstantsHolder.EXPRESSION_SUFFIX;
+        long count = factModelTree.getSimpleProperties().keySet().stream().filter(key -> key.startsWith(expressionName)).count();
+        factModelTree.addSimpleProperty(count > 0 ? expressionName + "_" + count : expressionName, ConstantsHolder.EXPRESSION_CLASSNAME);
```

Comment:
@yesamer 
Please, do not do that, and follow what we agreed upon: a fixed static part (ScesimExpression) followed by "-" and a  randomly generated two digit number. 
Beside that, for each "Object" there can be only one "expression" property, so that count has no meaning; if you need to take this in account, there is something wrong in the code

Reply:
Waiting with my re-review once this is addressed.

Reply:
@gitgabrio There is a meaning on the count. My idea was to count all property **manually added by the user** which start with the same name of "<FactName>Expression", in order to avoid any probability of collision. This was not intended to count the automatic generated expression type, I perfectly know there is only one of them. 
@jomarko I changed the behaviour, now the name of property is "<FactName>ScesimExpression" and it must be considered as a scenario keyword. This means, it shouldn't be used as a name for manually inserted properties.